modules/metrics/src/integrationTest/java/org/apache/ignite/internal/metrics/exporters/TestPushMetricExporter.java

```diff
@@ -36,23 +33,13 @@ public class TestPushMetricExporter extends PushMetricExporter<TestPushMetricsEx
 
     private static OutputStream outputStream;
 
-    private long period;
-
-    @Override
-    public void start(MetricProvider metricsProvider, TestPushMetricsExporterView configuration, Supplier<UUID> clusterIdSupplier,
-            String nodeName) {
-        period = configuration.period();
-
-        super.start(metricsProvider, configuration, clusterIdSupplier, nodeName);
-    }
-
     public static void setOutputStream(OutputStream outputStream) {
         TestPushMetricExporter.outputStream = outputStream;
     }
 
     @Override
     protected long period() {
-        return period;
+        return configuration().period();
```

Comment:
Because configuration in ancestor PushMetricExporter has type ExporterView which is common for push and pull exporters and doesn't contain period. At the same time configuration framework doesn't allow to create base class for PushExporterConfiguration as it is a polymorphic configuration.